File: app/src/main/java/com/bilulo/androidtest04/utils/ValidationUtil.java
package com.bilulo.androidtest04.utils;

import java.util.InputMismatchException;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class ValidationUtil {

    public static boolean isValidCpf(String cpf) {
        if (!isValidString(cpf) || !isFullCpf(cpf)) {
            return false;
        }
        if (cpf.contains(".") && cpf.contains("-")) {
            cpf = cpf.replace(".", "").replace("-", "");
        }
        if (cpf.equals("00000000000") ||
                cpf.equals("11111111111") ||
                cpf.equals("22222222222") || cpf.equals("33333333333") ||
                cpf.equals("44444444444") || cpf.equals("55555555555") ||
                cpf.equals("66666666666") || cpf.equals("77777777777") ||
                cpf.equals("88888888888") || cpf.equals("99999999999") ||
                (cpf.length() != 11))
            return (false);

        char dig10, dig11;
        int sm, i, r, num, peso;

        try {
            sm = 0;
            peso = 10;
            for (i = 0; i < 9; i++) {
                num = (int) (cpf.charAt(i) - 48);
                sm = sm + (num * peso);
                peso = peso - 1;
            }
            r = 11 - (sm % 11);
            if ((r == 10) || (r == 11))
                dig10 = '0';
            else dig10 = (char) (r + 48);

            sm = 0;
            peso = 11;
            for (i = 0; i < 10; i++) {
                num = (int) (cpf.charAt(i) - 48);
                sm = sm + (num * peso);
                peso = peso - 1;
            }
            r = 11 - (sm % 11);
            if ((r == 10) || (r == 11))
                dig11 = '0';
            else dig11 = (char) (r + 48);

            return (dig10 == cpf.charAt(9)) && (dig11 == cpf.charAt(10));
        } catch (InputMismatchException erro) {
            return (false);
        }
    }

    public static boolean isValidEmail(String email) {
        return isValidString(email) && android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches();
    }

    public static boolean isValidString(String string) {
        return string != null && !string.isEmpty();
    }

    public static boolean isFullCpf(String cpf) {
        return cpf.matches("[0-9]{3}\\.?[0-9]{3}\\.?[0-9]{3}-?[0-9]{2}");
    }

    public static boolean hasUppercase(String string) {
        return !string.equals(string.toLowerCase());
    }

    public static boolean hasSpecialCharacter(String string) {
        return !string.matches("[A-Za-z0-9 ]*");
    }

    public static boolean hasAlphanumericCharacter(String string) {
        Pattern p = Pattern.compile("[a-zA-Z0-9]");
        Matcher m = p.matcher(string);
        return m.find();
    }
}

File: app/src/main/java/com/br/rafael/TesteAndroidv2/Util/ValidationUtil.kt
package com.br.rafael.TesteAndroidv2.Util

import java.util.*
import java.util.regex.Pattern

class ValidationUtil {

    companion object{

        fun isCPF(doc: String): Boolean {

           var CPF = doc.replace(".", "").replace("-", "")

            if (CPF == "00000000000" || CPF == "11111111111" || CPF == "22222222222" || CPF == "33333333333" || CPF == "44444444444" || CPF == "55555555555" || CPF == "66666666666" || CPF == "77777777777" || CPF == "88888888888" || CPF == "99999999999" || CPF.length != 11)
                return false

            val dig10: Char
            val dig11: Char
            var sm: Int
            var i: Int
            var r: Int
            var num: Int
            var peso: Int

            try {
                // Calculo do 1o. Digito Verificador
                sm = 0
                peso = 10
                i = 0
                while (i < 9) {
                    // converte o i-esimo caractere do CPF em um numero:
                    // por exemplo, transforma o caractere '0' no inteiro 0
                    // (48 eh a posicao de '0' na tabela ASCII)
                    num = CPF[i].toInt() - 48
                    sm = sm + num * peso
                    peso = peso - 1
                    i++
                }

                r = 11 - sm % 11
                if (r == 10 || r == 11)
                    dig10 = '0'
                else
                    dig10 = (r + 48).toChar() // converte no respectivo caractere numerico

                // Calculo do 2o. Digito Verificador
                sm = 0
                peso = 11
                i = 0
                while (i < 10) {
                    num = CPF[i].toInt() - 48
                    sm += num * peso
                    peso -= 1
                    i++
                }

                r = 11 - sm % 11
                if (r == 10 || r == 11)
                    dig11 = '0'
                else
                    dig11 = (r + 48).toChar()

                return dig10 == CPF[9] && dig11 == CPF[10]
            } catch (erro: InputMismatchException) {
                return false
            }

        }

        fun isValidEmail(target: CharSequence?): Boolean {
            return if (target == null) false else android.util.Patterns.EMAIL_ADDRESS.matcher(target).matches()

        }

        fun validationPassword(password : String) : Boolean {


            return if(password.length >= 2) {
                val letter = Pattern.compile("[A-z]")
                val special = Pattern.compile("[!@#$%&*()_+=|<>?{}\\[\\]~-]")


                var hasLetter = letter.matcher(password)
                var hasSpecial = special.matcher(password)

                hasLetter.find() && hasSpecial.find()
            } else {
                false
            }


        }
    }
}